internal/routing/acquire.go

```diff
@@ -15,11 +16,27 @@ func acquire(w http.ResponseWriter, r *http.Request, ls *lockservice.SimpleLockS
 		return
 	}
 
-	desc := &lockservice.SimpleDescriptor{
-		FileID: string(body),
+	var req interface{}
+	err = json.Unmarshal(body, &req)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
 	}
 
-	err = ls.Acquire(desc)
+	data := req.(map[string]interface{})
+
+	for k, v := range data {
+		switch v := v.(type) {
+		case string:
+			if k == "FileID" {
+				desc := &lockservice.SimpleDescriptor{
+					FileID: string(v),
+				}
+				err = ls.Acquire(desc)
```

Comment:
Errors must be handled as soon as they're generated and not later elsewhere.

Reply:
This issue doesn't exist anymore as this code segment no longer exists.